### ISSUE TITLE
proxmox_virtual_environment_file times out inside Docker

### ISSUE THREAD

**Issue by omusavi** [bpg/terraform-provider-proxmox#604]
**Describe the bug**
I use `proxmox_virtual_environment_file` to upload an Ubuntu cloud image to the proxmox node so I can create VMs with it. I was planning on running terraform within the context of Docker via Github Actions and through local docker images (for initial bootstrap of cluster). I noticed that this step consistently fails via timeout when running inside Docker, but works as expected outside of docker. 

This same issue does not reproduce when creating a file using `source_raw`, so my cloud-init files work just fine with `proxmox_virtual_environment_file`, but downloading the image onto the node does not work. Adding the image manually to the node via the UI also works fine.

I have created a repo that demonstrates this failure: https://github.com/omusavi/tf-proxmox-docker

**To Reproduce**
1. `git clone https://github.com/omusavi/tf-proxmox-docker`
1. Run `docker build . -t tf-proxmox-docker`
2. Run `docker run -it tf-proxmox-docker`
3. Fill in terraform variables to point to your proxmox server
4. Type `yes` to apply

**Expected behavior**
Ubuntu image is uploaded to local storage (usually takes less than 2 minutes)

**Screenshots**

**Additional context**
- Single or clustered Proxmox: Single
- Provider version (ideally it should be the latest version): 0.33.0 (latest)
- Terraform version: latest
- OS (where you run Terraform from): Ubuntu 
- Debug logs (`TF_LOG=DEBUG terraform apply`): see in container, nothing obvious there

```
2023-10-05T14:45:04.275Z [DEBUG] provider.terraform-provider-proxmox_v0.33.0: Downloading file from URL: @caller=github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/file.go:378 tf_mux_provider=tf5to6server.v5tov6Server tf_resource_type=proxmox_virtual_environment_file @module=proxmox tf_provider_addr=registry.terraform.io/bpg/proxmox tf_req_id=e1ad074e-d8af-2244-711d-cd231fa7c982 tf_rpc=ApplyResourceChange url=https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img timestamp=2023-10-05T14:45:04.275Z
proxmox_virtual_environment_file.image: Still creating... [10s elapsed]
proxmox_virtual_environment_file.image: Still creating... [20s elapsed]
proxmox_virtual_environment_file.image: Still creating... [30s elapsed]
proxmox_virtual_environment_file.image: Still creating... [40s elapsed]
proxmox_virtual_environment_file.image: Still creating... [50s elapsed]
proxmox_virtual_environment_file.image: Still creating... [1m0s elapsed]
proxmox_virtual_environment_file.image: Still creating... [1m10s elapsed]
proxmox_virtual_environment_file.image: Still creating... [1m20s elapsed]

...
```

**Comment by bpg** [comment]
Hi @omusavi! 👋🏼 

Thanks for the report! It is interesting use case, definitely not one of what I've tested 😅

I think there is something in your environment that prevents the provider from downloading the file. 
Thanks for the demo repo, btw, it really helps! So, using your example as-is, I managed to get a little bit further in my environment, up until upload to PVE, but it failed, and I'm not sure why:
<img width="1201" alt="Screenshot 2023-10-05 at 9 48 51 PM" src="https://github.com/bpg/terraform-provider-proxmox/assets/627562/4c0344a3-883f-4725-b506-a32f5b15a522">

`signal: killed` ??

**Comment by omusavi** [comment]
That's a good point about the download vs upload to PVE...to narrow it down a bit, i added downloading of the img file in the `Dockerfile` so the file would be available already and passed the local path to the terraform script, and the same hang happened, so you're right its definitely happening at that stage. I did not get the same error as you either, but it failed in the same area (mine just hangs for a while and then spews a bunch of junk into the console...).

I just tested with a smaller file and it worked just fine, so its definitely something to do with the larger file. I tried a slightly smaller image at 421MB and it still happened. I'll see how I can narrow down further...

**Comment by bpg** [comment]
Ah, that makes sense. Provider buffers the whole file on disk in /tmp, you'd need at least 2x of the file size available there. I guess docker allocates quite a small ephemeral storage for your container. You can try bind mount a local folder when you start it, something like `docker run -it --mount type=bind,source=`pwd`,destination=/tmp tf-proxmox-docker `

**Comment by omusavi** [comment]
After playing around I think you are right, this is an environment issue. Unfortunately I don't have a real "solution".

I was running these containers within a WSL2 environment. Inside WSL2, everything was working fine, but in a docker container in WSL, throughput was insanely low. I was able to confirm this by watching `/var/tmp` on the PVE node for how fast the file was being uploaded. From docker-inside-WSL2, it was insanely slow, so it wasn't that nothing was happening, just that it was timing out long before it was finished. From WSL2, I was able to see how quickly the file transmitted.

I repeated the experiment on an Ubuntu machine I have, and working inside docker worked. Slower than the host, but not significantly. 

So clearly something is going on with WSL throughput to the PVE server. Maybe its just too many network hops - either way not something I care to investigate right now. Workaround is to not use docker insider WSL2!

Thanks for your help in investigating! And for the provider, its super helpful
